### ISSUE TITLE
Add IsFinal and IsNotFinal expression

### ISSUE THREAD

**Issue by loic425** [phparkitect/arkitect#240]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | yes

#### Summary

On Sylius architecture, we'd like to ensure our data fixtures factories are not final.

```php
$rules[] = Rule::allClasses()
    ->that(new Extend(ModelFactory::class))
    ->should(new IsNotFinal())
    ->because('Factories should be extensible')
;
```

**Comment by AlessandroMinoccheri** [comment]
Thanks @loic425 for the feature requested, I am working on this here: #241

**Comment by AlessandroMinoccheri** [comment]
Closed by #241
